paasta_tools/marathon_tools.py

```diff
@@ -1068,16 +1069,17 @@ def wait_for_app_to_launch_tasks(client, app_id, expected_tasks, exact_matches_o
     :param exact_matches_only: a boolean indicating whether we require exactly expected_tasks to be running
     """
     found = False
-    while not found:
-        try:
-            found = app_has_tasks(client, app_id, expected_tasks, exact_matches_only)
-        except NotFoundError:
-            pass
-        if found:
-            return
-        else:
-            print "waiting for app %s to have %d tasks. retrying" % (app_id, expected_tasks)
-            sleep(0.5)
+    with requests_cache.disabled():
```

Comment:
I was hoping libraries like marathon_tools do not control caching behavior. Applications (setup_marathon_job, paasta_serviceinit etc) at a higher level are the ones that setup cache if needed, and disables cache temporarily. If the application does not setup cache, there is no need to disable cache at all. I would suggest move requests_cache.disabled up to the application that fails on caching.